File: 0/Templates/CSharp/ProjectTemplates/NavigationAppTemplate/UI/Converters/NavigationItemToObjectConverter.cs
using $safeprojectname$.Presentation;

namespace $safeprojectname$.UI.Converters
{
    /// <summary>
    /// Converts a <see cref="NavigationItem"/> instance to object and vice versa.
    /// </summary>
    public class NavigationItemToObjectConverter
        : ValueConverter<NavigationItem, object>
    {
        /// <summary>
        /// Converts a source value to the target type.
        /// </summary>
        /// <param name="value"></param>
        /// <param name="parameter"></param>
        /// <param name="language"></param>
        /// <returns></returns>
        protected override object Convert(NavigationItem value, object parameter, string language)
        {
            return value;
        }

        /// <summary>
        /// Converts a target value back to the source type.
        /// </summary>
        /// <param name="value"></param>
        /// <param name="parameter"></param>
        /// <param name="language"></param>
        /// <returns></returns>
        protected override NavigationItem ConvertBack(object value, object parameter, string language)
        {
            return (NavigationItem)value;
        }
    }
}

File: 0/Templates/VisualBasic/ProjectTemplates/NavigationAppVBTemplate/UI/Converters/NavigationItemToObjectConverter.vb
Imports $safeprojectname$.Presentation

Namespace UI.Converters
    ''' <summary>
    ''' Converts a <see cref="NavigationItem"/> instance to object and vice versa.
    ''' </summary>
    Public Class NavigationItemToObjectConverter
        Inherits ValueConverter(Of NavigationItem, Object)

        Public Overrides Function Convert(value As NavigationItem, parameter As Object, language As String) As Object
            Return value
        End Function

        Public Overrides Function ConvertBack(value As Object, parameter As Object, language As String) As NavigationItem
            Return TryCast(value, NavigationItem)
        End Function
    End Class
End Namespace

